R/GeospatialFunctions.R

```diff
@@ -0,0 +1,607 @@
+    
+  }
+  
+  all_features <- dplyr::bind_rows(all_features)
+  
+  return(all_features)
+  
+}
+
+
+#' TADA_GetATTAINS
+#' 
+#' Link catchment-based ATTAINS assessment unit data to Water Quality Portal observations, often imported via `TADA_DataRetrieval()`. This function returns the same raw objects that are mapped in `TADA_ViewATTAINS()`.
+#' 
+#' @param data A dataframe created by `TADA_DataRetrieval()` or the sf equivalent made by `TADA_MakeSpatial()`.
+#' @param return Whether to return the associated ATTAINS_catchments, ATTAINS_lines, ATTAINS_points, and ATTAINS_polygons shapefile objects. TRUE (yes, return) or FALSE (no, do not return). All ATTAINS features are in WGS84 (crs = 4326).
+#' 
+#' @return A modified `TADA_DataRetrieval()` dataframe with additional columns associated with the ATTAINS assessment unit data. Or, if return = TRUE, a list containing that same data frame plus the raw ATTAINS features associated with those observations.
+#' 
+#' @seealso [TADA_DataRetrieval()]
+#' @seealso [TADA_MakeSpatial()]
+#' @seealso [TADA_ViewATTAINS()]
+#' 
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#'tada_data <- TADA_DataRetrieval(startDate = "1990-01-01",
+#'                                endDate = "1995-12-31",
+#'                                characteristicName = "pH",
+#'                                statecode = "NV",
+#'                                applyautoclean = TRUE)
+#'
+#'tada_attains <- TADA_GetATTAINS(data = tada_data, return = FALSE)
+#'tada_attains_list <- TADA_GetATTAINS(data = tada_data, return = TRUE)
+#' }
+
+TADA_GetATTAINS <- function(data, return = TRUE){
+  
+  
+  attains_names <- c(
+    "ATTAINS.organizationid", "ATTAINS.submissionid", "ATTAINS.hasprotectionplan",
+    "ATTAINS.assessmentunitname", "ATTAINS.nhdplusid", "ATTAINS.tas303d",                                                  
+    "ATTAINS.isthreatened", "ATTAINS.state", "ATTAINS.on303dlist",                                               
+    "ATTAINS.organizationname", "ATTAINS.region", "ATTAINS.Shape_Length",                                             
+    "ATTAINS.reportingcycle", "ATTAINS.assmnt_joinkey", "ATTAINS.hastmdl",                                                  
+    "ATTAINS.orgtype", "ATTAINS.permid_joinkey", "ATTAINS.catchmentistribal",                                        
+    "ATTAINS.ircategory", "ATTAINS.waterbodyreportlink", "ATTAINS.assessmentunitidentifier",                                 
+    "ATTAINS.overallstatus", "ATTAINS.isassessed", "ATTAINS.isimpaired",                                               
+    "ATTAINS.has4bplan", "ATTAINS.huc12", "ATTAINS.hasalternativeplan",                                      
+    "ATTAINS.visionpriority303d", "ATTAINS.areasqkm", "ATTAINS.catchmentareasqkm",                                       
+    "ATTAINS.catchmentstatecode", "ATTAINS.catchmentresolution", "ATTAINS.Shape_Area"
+  )
+  
+  if (all(attains_names %in% colnames(data))){
```

Comment:
Comment, non-blocking: Maybe this should be "any"? Just thinking if someone has reduced columns they might not have retained all of the columns?

Reply:
Done!